src/lib/histogram.lua

```diff
@@ -213,8 +259,25 @@ function wrap_thunk(histogram, thunk, now)
    end
 end
 
+function iterate(histogram)
+   print("histogram is", histogram, histogram.growth_factor_log)
+   local lo = histogram.minimum
+   local count = histogram.count
+   local hi = lo * (math.exp(histogram.growth_factor_log)^(num_buckets - 2))
+   local entry = 0
+   local max_entry = num_buckets
+   local function next_entry()
+      if entry < max_entry then
+         entry = entry + 1
+         return histogram.buckets[entry - 1], lo, hi, count
+      end
+   end
+   return next_entry
+end
```

Comment:
I think having to peek into the buckets using an index is a bad design default, hence the difference - I originally returned the index as well, but didn't like what it did to the code that used the iterator. I ended up returning both the value in the bucket and, secondarily, the index - to allow the pattern where it's being compared with a previous bucket with minimal boilerplate or redesign. On the ugly side, the value and index are in the opposite order from ipairs... :-)
In general, I like my iterators to give me values rather than an index to get a value - or both.
I'll look at the iterate function in 795.